RecoVertex/V0Producer/src/V0Fitter.h

```diff
@@ -47,18 +41,28 @@
 #include "Geometry/CommonDetUnit/interface/GeomDet.h"
 #include "Geometry/TrackerGeometryBuilder/interface/GluedGeomDet.h"
 #include "FWCore/Framework/interface/ConsumesCollector.h"
-
 #include "DataFormats/TrackingRecHit/interface/TrackingRecHit.h"
 #include "DataFormats/BeamSpot/interface/BeamSpot.h"
-
-#include <string>
-#include <fstream>
+#include "CommonTools/CandUtils/interface/AddFourMomenta.h"
```

Comment:
a somewhat related question:
if I notice that the code compiles WITHOUT a certain include statement,
such as
#include "FWCore/MessageLogger/interface/MessageLogger.h"
is it generally safe to leave out entirely?


On Tue, Oct 6, 2015 at 3:23 PM, Slava Krutelyov <notifications@github.com>
wrote:

> In RecoVertex/V0Producer/src/V0Fitter.h
> <https://github.com/cms-sw/cmssw/pull/11660#discussion_r41309868>:
>
> >  #include "DataFormats/TrackingRecHit/interface/TrackingRecHit.h"
> >  #include "DataFormats/BeamSpot/interface/BeamSpot.h"
> > -
> > -#include <string>
> > -#include <fstream>
> > +#include "CommonTools/CandUtils/interface/AddFourMomenta.h"
>
> I didn't mean to suggest removal of all includes: only most of the new
> ones.
>
> The diffs in this file have only one new non-trivial member change:
> reco::Vertex, which is already included.
>
> All new includes, like MessageLogger, TrajectoryStateTransform,
> TSCBLBuilderNoMaterial don't appear to be necessary in the class interface
> declaration and should be moved to the implementation file.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/11660/files#r41309868>.
>